### PR TITLE
#003 Fixing ER models

### PR DESCRIPTION
**Pull request by frankflej**
## What does this PR do?

This PR fixes the project entity relationship models, improves query responses, and renames API endpoints.

## Description of Task to be completed?

The task aims to correcting the existing ER Model to add necessary relationships between models/entities, and improve queries and responses to include informations that make sense to the recipient. The task also involves renaming the project's API Endpoints for uniformity.

## How should this be manually tested?

- Clone the repository using `git clone` [link](https://github.com/atlp-rwanda/e-comm-team-techsmith-bn)

- Checkout to the branch using `git checkout bg-fix-relations-model-003`

- Install node packages that the application depends on: `npm i`

- Create a `.env` file in the root directory and add all environment variables required to run the application. The required variables are listed in the `.env.example` file available in root directory of the application.

- Start the application development server: `npm run dev`

## What are the relevant pivotal tracker/Trello stories?

- [#184752187](https://www.pivotaltracker.com/story/show/184752187)

## Screenshots (if appropriate)
better-responses

### CODE REVIEW

**Review by bergad37 — COMMENTED** [review]
The work is done well,and associations have been added.It would be safe that on the admin routes `verifyIsAdmin` middleware should be included to access `getUsers` controller.

**Comment by nishimweprince** [comment]
> The work is done well,and associations have been added.It would be safe that on the admin routes `verifyIsAdmin` middleware should be included to access `getUsers` controller.

Thank you, I have made the changes.

**Review by bergad37 — APPROVED** [review]
Well done!

**Review by nishimweprince — APPROVED** [review]
Excellent work, it is also great that the endpoints follow a uniform naming convention.

File: src/controllers/wishlistController.js
Comment (Dyves1) [inline src/controllers/wishlistController.js:1]: this name is perfect

**Review by Dyves1 — APPROVED** [review]
That works nice since you added the associations which will be so helpful in retrieving data easily than first getting in the two entities differently

**Review by jcharles921 — APPROVED** [review]
Thanks for adding and fixing those relationships, it is a great work the team did.

**Review by niyontwali — APPROVED** [review]
Thanks guys for this. I know it was a bit challenging to refine the models but trust me this will be easy for you to fetch data and most specific relational data

**Pull request merged**